### PR TITLE
Issue 75

### LINKED ISSUE #75: allow  tqdm progress bar to be used
the parser might get a progress bar so that the parsing progress may be shown

### PR DESCRIPTION
**Pull request by WolfgangFahl**
fixes #75

### CODE REVIEW

File: quiffen/__init__.py [hunk 1]
```diff
@@ -18,4 +18,9 @@
 from quiffen.core.qif import ParserException, Qif, QifDataType
 from quiffen.core.security import Security
 from quiffen.core.split import Split
-from quiffen.core.transaction import Transaction, TransactionLike, TransactionList
+from quiffen.core.transaction import (
+    Transaction,
+    TransactionConfig,
+    TransactionLike,
+    TransactionList,
+)
```
Comment (isaacharrisholt) [inline quiffen/__init__.py:26]: This seems to have code from other open PRs. Please separate this so it only contains the relevant code.

File: quiffen/core/qif.py [hunk 2]
```diff
@@ -109,6 +109,7 @@ def parse(
         separator: str = "\n",
         day_first: bool = False,
         encoding: str = "utf-8",
+        progress=None,
```
Comment (isaacharrisholt) [inline quiffen/core/qif.py:112]: What type is `progress` here? What's it supposed to represent? It's not immediately clear from the naming.

**Review by isaacharrisholt — COMMENTED** [review]
Let's discuss a solution in #75. I'm not 100% sure this is actually a feature I believe should be included in Quiffen.

File: quiffen/__init__.py (hunk 1 above)
Comment (WolfgangFahl) [reply quiffen/__init__.py:26]: progress is a tqdm or ducktype compatible progress  bar. For large imports having the progress bar is IMHO a very important feature and the implementation is quite non invasive.

Comment (isaacharrisholt) [reply quiffen/__init__.py:26]: Then it needs to have a type hint as such, at the very least. Again, this is another feature that needs discussion, as it has long-term implications for me in terms of maintenance.

**Comment by isaacharrisholt** [comment]
Closing due to inactivity. Feel free to reopen, but keep code in separate PRs where possible to make things easier to review.

**Pull request closed without merging**